lib/indicadoresEconomicosBCCR.js

```diff
@@ -1,43 +1,63 @@
 'use strict';
 
-var axios = require('axios');
-var querystring = require('querystring');
-var DOMParser = require('xmldom').DOMParser;
-
-module.exports.default = indicadoresEconomicosBCCR;
-module.exports = indicadoresEconomicosBCCR;
-
-function indicadoresEconomicosBCCR(email, token, fechaInicio, fechaFinal) {
-    try {
-        var todayDate = new Date();
-        var BCCRurl = 'https://gee.bccr.fi.cr/Indicadores/Suscripciones/WS/wsindicadoreseconomicos.asmx/ObtenerIndicadoresEconomicos';
-        var payload = {
-            FechaInicio: fechaInicio ? fechaInicio : todayDate.getDate() + "/" + (todayDate.getMonth() + 1) + "/" + todayDate.getFullYear(),
-            FechaFinal: fechaFinal ? fechaFinal : todayDate.getDate() + "/" + (todayDate.getMonth() + 1) + "/" + todayDate.getFullYear(),
-            Nombre: 'N',
-            SubNiveles: 'N',
-            Indicador: 317,
-            CorreoElectronico: email,
-            Token: token,
-        };
-        var postCompra = axios.post(BCCRurl, querystring.stringify(payload));
-
-        payload.Indicador = 318;
-
-        var postVenta = axios.post(BCCRurl, querystring.stringify(payload));
-
-        return axios.all([postCompra, postVenta]).then(axios.spread(function (compra, venta) {
-            var compraNode = new DOMParser().parseFromString(compra.data, 'text/xml');
-            var ventaNode = new DOMParser().parseFromString(venta.data, 'text/xml'); 
-
-            return {
-                'compra': Math.pow(parseFloat(compraNode.documentElement.getElementsByTagName('NUM_VALOR')[0].childNodes[0].nodeValue).toFixed(2), 1),
-                'venta': Math.pow(parseFloat(ventaNode.documentElement.getElementsByTagName('NUM_VALOR')[0].childNodes[0].nodeValue).toFixed(2), 1)
-            };
-        }));
-
-    } catch (error) {
-        throw new Error(error);
+const axios = require('axios');
+const querystring = require('querystring');
+const DOMParser = require('xmldom').DOMParser;
+const validateDate = (date) => {
+    console.log(date)
+    const dateRegex = /(((0|1)[0-9]|2[0-9]|3[0-1])\/(0[1-9]|1[0-2])\/((19|20)\d\d))$/;
+    return dateRegex.test(date)
+}
+
+const getFormattedTodayDate = () => {
+    const now = new Date();
+    return `${now.getDate().toString().padStart(2, "0")}/${(now.getMonth() + 1).toString().padStart(2, "0")}/${now.getFullYear()}`
+}
+
+const indicadoresEconomicosBCCR = (email, token, desde = getFormattedTodayDate(), hasta = getFormattedTodayDate(), indicadores = [317, 318], nombre = 'N', subNiveles = 'N') => {
+    switch (true) {
```

Comment:
porque usar esto? me parece que querías agregar Guard Clauses

Reply:
Para primero, prevenir enviar una petición al web service si el formato u otros datos son incorrectos, y segundo utilice un switch para evitar anidar muchos ifs, y que quede abierto a más posibles casos que deban retornar un error antes de enviar la petición. 

Como lo mejorarías en este caso?

Reply:
En lugar de ese switch, yo prefiero utilizar Yup o poner esa misma validation pero ambas en un function aparte, y no dentro de indicadoresEconomicosBCCR. Lo indico como sugerencia, que opinan?

Reply:
Tiene mucho sentido lo de poner la validación fuera de la función.

YUP nunca lo he utilizado, así que posiblemente optaría por sacar la validación de la función principal, quizás en otro archivo como un helper, y luego importarlo.

Tienes algún ejemplo de cómo sería con YUP? Vi un package que se llama así, no se si sea necesario agregar otra dependencia.

Reply:
no creo que sea necesario otra dependencia, ya que la validación la tienes lista. Pero si lo agregaría como un function separada.

Reply:
Bien, he separado las validaciones en un archivo aparte, haciendo esto noté que cuando las credenciales no existen, el webservice devuelve un codigo 200, por lo que aún cuando hay un error (ej: las credenciales no existen), el catch de axios nunca va a ejecutarse, por tanto agregué otra validación de la respuesta, devolviendo un console.error() con un mensaje de error ya que al estar dentro del .then del axios si hago throw error, sucede un error ya que se finalizó la promesa con un error dentro del try... no se si me explico, el punto es que para darle un feedback al usuario simplemente devuelvo un mensaje, ya que las validaciones que se hicieron cubren practicamente la mayoría de las posibles fallas, esta validación de la respuesta, podría solo corresponder a credenciales, o en un caso especial a un error del servidor del webservice... si tienen alguna sugerencia para solucionar esto, es bienvenida.